src/plots/cartesian/select.js

```diff
@@ -308,18 +276,24 @@ function prepSelect(e, startX, startY, dragOptions, mode) {
 
         throttle.done(throttleID).then(function() {
             throttle.clear(throttleID);
+
             if(numClicks === 2) {
-                // clear selection on doubleclick
-                outlines.remove();
                 for(i = 0; i < searchTraces.length; i++) {
                     searchInfo = searchTraces[i];
-                    searchInfo._module.selectPoints(searchInfo, false);
+                    searchInfo._module.toggleSelected(searchInfo, false);
                 }
 
                 updateSelectedState(gd, searchTraces);
+
+                // clear visual boundaries of selection area if displayed
+                outlines.remove();
```

Comment:
Why did you move this after `updateSelectedState`?

Reply:
TBH, I'm not sure. Fun fact: I even had it beneath emitting `plotly_selected` event which then broke some unit tests that assume outline elements are no longer in DOM when `plotly_selected` event is received.

But why did I really moved it below `updateStelectedState`? As far as I can remember, in the early commits, I thought I could move the line before the `if(numClicks === 2)` condition. For some reason that wasn't correct and I moved it back into the if block. So I guess no rationale behind it other than me not putting it back where it always should have been.